src/main/java/reposense/util/FileUtil.java

```diff
@@ -55,55 +64,81 @@ public static void deleteDirectory(String root) throws IOException {
         }
     }
 
-    public static void unzip(ZipInputStream zipInput, String destinationFolder) {
-        Path directory = Paths.get(destinationFolder);
-
-        // buffer for read and write data to file
-        byte[] buffer = new byte[2048];
+    /**
+     * Zips all the files of {@code fileTypes} contained in {@code sourceAndOutputPath} directory into the same folder.
+     */
+    public static void zip(Path sourceAndOutputPath, String... fileTypes) {
+        FileUtil.zip(sourceAndOutputPath, sourceAndOutputPath, fileTypes);
+    }
 
-        try {
-            Files.createDirectories(directory);
-            ZipEntry entry = zipInput.getNextEntry();
+    /**
+     * Zips all the {@code fileTypes} files contained in the {@code sourcePath} and its subdirectories.
+     * Creates the zipped {@code ZIP_FILE} file in the {@code outputPath}.
+     */
+    public static void zip(Path sourcePath, Path outputPath, String... fileTypes) {
+        try (
+                FileOutputStream fos = new FileOutputStream(outputPath + File.separator + ZIP_FILE);
+                ZipOutputStream zos = new ZipOutputStream(fos)
+        ) {
+            Set<Path> allFiles = getFilePaths(sourcePath, fileTypes);
+            for (Path path : allFiles) {
+                try (InputStream is = Files.newInputStream(path)) {
+                    if (Files.isDirectory(path)) {
+                        zos.putNextEntry(new ZipEntry(sourcePath.relativize(path.toAbsolutePath()).toString()
+                                + File.separator));
+                        continue;
+                    }
+                    zos.putNextEntry(new ZipEntry(sourcePath.relativize(path.toAbsolutePath()).toString()));
+                    int length;
+                    while ((length = is.read(buffer.array())) > 0) {
+                        zos.write(buffer.array(), 0, length);
+                    }
+                }
+            }
+        } catch (IOException ioe) {
+            logger.log(Level.SEVERE, ioe.getMessage(), ioe);
+        }
+    }
```

Comment:
Suggested fix to Windows as follows:

```
    public static void zip(Path sourcePath, Path outputPath, String... fileTypes) {
        try (
                FileOutputStream fos = new FileOutputStream(outputPath + File.separator + ZIP_FILE);
                ZipOutputStream zos = new ZipOutputStream(fos)
        ) {
            Set<Path> allFilePaths = getFilePaths(sourcePath, fileTypes);
            for (Path filePath : allFilePaths) {
                String path = sourcePath.relativize(filePath.toAbsolutePath()).toString();
                String zipEntryPath = Files.isDirectory(filePath) ? path + "/" : path;
                zos.putNextEntry(new ZipEntry(zipEntryPath.replace("\\", "/")));

                if (Files.isRegularFile(filePath)) {
                    byte[] data = Files.readAllBytes(filePath);
                    zos.write(data, 0, data.length);
                }

                zos.closeEntry();
            }
        } catch (IOException ioe) {
            logger.log(Level.SEVERE, ioe.getMessage(), ioe);
        }
    }
```

Reply:
Verified on my Windows 10 side as well 👍

Reply:
I don't think we need `InputStream` to read a file in byte as we are dealing with very small files.

Reply:
>I don't think we need InputStream to read the bytes of a file as we are dealing with very small files

Erm, some of the `.json` files can go up to several 100k LOC (our `authorship.json` for `testrepo-charlie` is at 3739KB at the moment, will increase in other larger projects), so we should probably shift back to `InputStream` and read using a buffer.

Reply:
Ah... let's keep the `InputStream` 👍